sklearn/cluster/optics_.py

```diff
@@ -584,8 +640,9 @@ def _automatic_cluster(reachability_plot, ordering,
     neighborhood_size = int(min_maxima_ratio * len(ordering))
 
     # Should this check for < min_samples? Should this be public?
-    if min_cluster_size < 5:
-        min_cluster_size = 5
+    # why 5? shouldn't it be 2?
```

Comment:
you're right, the 5 is strange. 2 is logically a minimum...

In DBSCAN, I don't think min_samples is the minimum cluster size, as a core point will include all its min_samples neighborhood only if all its samples were not within `eps` of another core point that was traversed earlier.

I am not certain if the same applies here.

2 seems a good place to start without other justification, when we already have a parameter to control minimum cluster size.

What I might consider is ditching `min_cluster_size_ratio` and just making it `min_cluster_size` (it's more a fraction than a ratio anyway).